renderdoc/driver/vulkan/wrappers/vk_wsi_funcs.cpp

```diff
@@ -621,15 +621,6 @@ VkResult WrappedVulkan::vkCreateSwapchainKHR(VkDevice device,
 
 VkResult WrappedVulkan::vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR *pPresentInfo)
 {
-  if(IsBackgroundCapturing(m_State))
-  {
-    RenderDoc::Inst().Tick();
-
-    GetResourceManager()->FlushPendingDirty();
-  }
-
-  m_FrameCounter++;    // first present becomes frame #1, this function is at the end of the frame
-
```

Comment:
Removing this code and putting it into the extracted function changes the order of operations. `m_FrameCounter` isn't incremented until the end of the function meaning the value displayed on the overlay will be off by 1.

The easiest solution is probably just to extract these into two separate functions to be called in the original locations here, then just one after the other in `vkQueuePresentKHR` above when doing a fake present.